specs/sharding/beacon-chain.md

```diff
@@ -448,15 +469,19 @@ def get_start_shard(state: BeaconState, slot: Slot) -> Shard:
 ```python
 def compute_shard_from_committee_index(state: BeaconState, slot: Slot, index: CommitteeIndex) -> Shard:
     active_shards = get_active_shard_count(state, compute_epoch_at_slot(slot))
+    assert index < active_shards
     return Shard((index + get_start_shard(state, slot)) % active_shards)
 ```
 
 #### `compute_committee_index_from_shard`
 
 ```python
 def compute_committee_index_from_shard(state: BeaconState, slot: Slot, shard: Shard) -> CommitteeIndex:
-    active_shards = get_active_shard_count(state, compute_epoch_at_slot(slot))
-    return CommitteeIndex((active_shards + shard - get_start_shard(state, slot)) % active_shards)    
+    epoch = compute_epoch_at_slot(slot)
+    active_shards = get_active_shard_count(state, epoch)
+    index = CommitteeIndex((active_shards + shard - get_start_shard(state, slot)) % active_shards)
+    assert index >= get_committee_count_per_slot(state, epoch)
```

Comment:
Are you sure we don't want `<` here instead of `<=`?

Reply:
Good catch, we should make it executable for testing to catch this earlier, I've been writing too much `if ... { return err }` Go code with the reverse assert expressions